### PR TITLE
python3Packages: update font related packages (fonttools, ufoprocessor, afdko, fontpens)

### PR DESCRIPTION
**Pull request by sternenseemann**
## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [25.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) (or backporting [24.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) and [25.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by sternenseemann** [comment]
Eval failure looks to be unrelated.

**Comment by sternenseemann** [comment]
Tested noto-fonts-emoji which works fine.

File: pkgs/development/python-modules/fontpens/default.nix [hunk 1]
```diff
@@ -1,21 +1,25 @@
...
 
-  src = fetchPypi {
-    pname = "fontPens";
-    inherit version;
-    sha256 = "1za15dzsnymq6d9x7xdfqwgw4a3003wj75fn2crhyidkfd2s3nd6";
-    extension = "zip";
+  src = fetchFromGitHub {
+    owner = "robotools";
+    repo = pname;
+    rev = "v${version}";
```
Comment (mweinelt) [inline pkgs/development/python-modules/fontpens/default.nix:17]:
```suggestion
    tag = "v${version}";
```

File: pkgs/development/python-modules/afdko/default.nix [hunk 2]
```diff
@@ -36,7 +36,7 @@ buildPythonPackage rec {
     owner = "adobe-type-tools";
     repo = "afdko";
     rev = "refs/tags/${version}";
```
Comment (mweinelt) [inline pkgs/development/python-modules/afdko/default.nix:38]:
```suggestion
    tag = version;
```

File: pkgs/development/python-modules/fonttools/default.nix [hunk 3]
```diff
@@ -36,7 +36,7 @@ buildPythonPackage rec {
     owner = "fonttools";
     repo = "fonttools";
     rev = "refs/tags/${version}";
```
Comment (mweinelt) [inline pkgs/development/python-modules/fonttools/default.nix:38]:
```suggestion
    tag = version;
```

File: pkgs/development/python-modules/ufoprocessor/default.nix [hunk 4]
```diff
@@ -14,32 +14,32 @@
...
   src = fetchPypi {
-    pname = "ufoProcessor";
-    inherit version;
-    sha256 = "0ns11aamgavgsfj8qf5kq7dvzmgl0mhr1cbych2f075ipfdvva5s";
-    extension = "zip";
+    inherit pname version;
+    sha256 = "1187g7xs6z8i2hzfkqhfd59qsdvzydqnmwhaz71nsi1zf5bw59gw";
   };
 
   nativeBuildInputs = [ setuptools-scm ];
```
Comment (mweinelt) [inline pkgs/development/python-modules/ufoprocessor/default.nix:25]:
```suggestion
  build-system = [ setuptools-scm ];
```

File: pkgs/development/python-modules/ufoprocessor/default.nix [hunk 5]
```diff
@@ -14,32 +14,32 @@
...
-  propagatedBuildInputs = [
-    defcon
-    lxml
-    fonttools
-    fs
-    fontmath
-    fontparts
-    mutatormath
-  ];
+  propagatedBuildInputs =
```
Comment (mweinelt) [inline pkgs/development/python-modules/ufoprocessor/default.nix:27]:
```suggestion
  dependencies =
```

**Review by mweinelt — COMMENTED** [review]
Looks fine to me, some minor recommendations.

**Comment by sternenseemann** [comment]
Thanks, addressed.

**Pull request merged**